src/UpdatedDate.js

```diff
@@ -10,19 +10,36 @@ export default function UpdatedDate(props) {
     "Friday",
     "Saturday",
   ];
-  let day = days[props.date.getDay()];
-  let hours = props.date.getHours();
-  let minutes = props.date.getMinutes();
+
+  props = props.date;
+
+  let correctedOffset = ((props.timeZoneOffset * 100) / 60 / 60);
+  correctedOffset = correctedOffset / 100;
+
+  let currentDate = (new Date().toISOString()).replace(/\.\d*Z/, "");
```

Comment:
I don't understand what's following the .replace. What does this expression mean?

Reply:
## Setup
The 1st argument of the .replace function can take a string literal or regex, in this case it is regex.
The second argument is the value with which I shall replace the first argument.

So I am replacing /\.\d*Z/ with "" aka "nothing" aka "make the value of the first argument disappear"

Regex helps you "match portions of a string". It is very powerful - though, it can be confusing at first (or always...rofl)

In our current case, I want to manipulate the ISO String created, such as "2020-10-15T05:45:57.361Z" and put it into the format of "2020-10-15T05:45:57"

### Step 1
The two forward slashes "contain" the regex:
//  <--
Everything inside of it determines the value (possible values) of the search term

The first part of what I want to match is a literal period/decimal sign
In regex syntax, a period is actually a symbol for 'generic character that can be found in a string'.
Period means "matches any character (except for line terminators)"

I don't want regex language there. I want the actual period - so I have to "escape" the period. Escaping a character is accomplished by putting a BACKWARD slash and then the keystroke one wishes to escape. So by typing
.\ <--- I get a 'real' 'literal' 'actual' period.

So we have:
/\./ <--- which matches any "." in the string I'm evaluating

### Step 2
\d <-- in regex syntax indicates "any single digit number from 0 to 9". One number only.
\d = one number = 2 || 6 || 9

So now we have:
/\.\d/ <-- which matches .3 || .9 etc
It does NOT match .a || .a5 etc

### Step 3
Next, are likely, more numbers.
How many? I don't know.
Could be two, could be five, could be fifteen hundred of them.
All I know is, there are numbers after the period and I don't want them (so I'm trying to 'match them' so I can do something to them)

In regex syntax: the star/asterisk aka * denotes "Matches between zero and unlimited times, as many times as possible" <-- of whatever individual item came right before it.

What came before the * ?
A singular number.

What does \d* mean?
It means "match 0 or more numbers in a row"

\d* matches 3 || 33 || 0123 || 230954823947040 || 1
How many numbers does it match? As many numbers as there are in a row.
What if there are no numbers? Oh well. But it matches ZERO to unlimited... so it doesn't care. That means the number here is also OPTIONAL

Aside: If we used the regex syntax denoted as a "plus sign" aka + we would HAVE to include at least one number. In contrast to *, the + means "Matches between one and unlimited times, as many times as possible"

So in total we have:

/\.\d*/  <-- which matches .123 || .33 || .3 || .1234567890098765432

### Step 4
Finally:
I know that what I'm searching for ends with the letter Z
It always ends with the letter Z.
That makes it easy.
I want Z. Z is static. Type Z :-P

So now I have:
/\.\d*Z/ <-- which matches .123Z || .33Z || .3Z || .1234567890098765432Z

Be careful to note, that it does NOT match any of the following .123 || .33z || .3S || .1234567890098765432

Lastly, extra pay attention that (this is just theory for you, so you know how it works):
if I am searching the string "2020-10-15T05:45:57.361ZYoMammaSoBlackWhenSheTookABathSheMadeTea"

/\.\d*Z/ <-- ONLY matches this part: ".361Z"
Not what comes before. Not what comes after.
This is exactly what I want.

I don't want nor need this information in the string.

The regex matched what I wanted - as my first argument - the search term.
My second argument is an empty string aka ""

So whatever my regex finds is replaced/written over with nothing, aka deleted.

Take that ISO zero UTC offset... or whatever it is called.